readme.md

```diff
@@ -98,17 +104,11 @@ When the automatic PR is created for azure-sdk-for-net if there are any issues f
   }
   ```
   4. Locate the `dll` file generated for the generator, usually it could be found `autorest.csharp` repository in this directory: `artifacts\bin\AutoRest.CSharp\Debug\net6.0\AutoRest.CSharp.dll`
-  5. Go to [this file](https://github.com/Azure/azure-sdk-for-net/blob/285cbdc343f01774b2b33e758662222bc8a3a33b/eng/common/scripts/Cadl-Project-Generate.ps1#L80) in your local `azure-sdk-for-net` repository, and change it like this by appending a new option to the emitter invocation:
-  ```diff
-  -$cadlCompileCommand = "npx cadl compile $mainCadlFile --emit $emitterName$emitterAdditionalOptions"
-  +$cadlCompileCommand = "npx cadl compile $mainCadlFile --emit $emitterName$emitterAdditionalOptions --option @azure-tools/cadl-csharp.csharpGeneratorPath=/absolute/path/to/artifacts/bin/AutoRest.CSharp/Debug/net6.0/AutoRest.CSharp.dll"
-  ```
-  This process right now is complicated, and the improvement is tracking in [this issue](https://github.com/Azure/azure-sdk-for-net/issues/34357).
 - At this point there are several ways to generate but the idea is to generate and test all projects that will be affected by your PR.  There are tools such as [this](https://github.com/ArcturusZhang/Regen) which are written to handle specific cases.  If you are unsure you should run against all projects to be safe.
 
 To regen and test everything in azure-sdk-for-net after you have updated to use your new local build do the following:
 
-- First generate all projects in the repo by executing `dotnet build [RepoRoot]/eng/service.proj /t:GenerateCode`
+- First generate all projects in the repo by executing `dotnet build [RepoRoot]/eng/service.proj /t:GenerateCode`. If your generator PR needs to apply to typespec/cadl projects, or your generator PR changes anything in our emitter (the `src\CADL.Extension\Emitter.Csharp` project), you will need to run `dotnet build [RepoRoot]/eng/service.proj /t:GenerateCode  /p:CadlAdditionalOptions="@azure-tools/cadl-csharp.csharpGeneratorPath=/absolute/path/to/artifacts/bin/AutoRest.CSharp/Debug/net6.0/AutoRest.CSharp.dll`.
```

Comment:
I think we should just have key/value pairs here and have the script wrap them into cadl syntax.

`dotnet build /t:GenerateCode /p:CadlAdditionalOptions=csharpGeneratorPath=/absolute/path/to/artifacts/bin/AutoRest.CSharp/Debug/net6.0/AutoRest.CSharp.dll`

Inside the script we should wrap this using the existing methods to get the emitter name.

Reply:
I also wonder if we should have something "simpler" like `dotnet build /t:GenerateCode /p:UseLocalBuild` which can look at the `emitter-package.json` file and get the local build path from there since it will be in the same location relatively?

Reply:
> I also wonder if we should have something "simpler" like `dotnet build /t:GenerateCode /p:UseLocalBuild` which can look at the `emitter-package.json` file and get the local build path from there since it will be in the same location relatively?

That would be very convenient, vote for that

Reply:
I should have thought about the name before making the suggestion but if we go that route it should be more precise like `UseLocalGenerator` or something like that.

Reply:
We may need to override other options beside 'csharpGeneratorPath'. And it is common library for all languages. There will be special option for other language to override.
So I think we shall give out a way to set options.

Reply:
> I also wonder if we should have something "simpler" like `dotnet build /t:GenerateCode /p:UseLocalBuild` which can look at the `emitter-package.json` file and get the local build path from there since it will be in the same location relatively?

I update to remove emitter name in the parameter, just need to set key-value pair.

Reply:
> I think we should just have key/value pairs here and have the script wrap them into cadl syntax.
> 
> `dotnet build /t:GenerateCode /p:CadlAdditionalOptions=csharpGeneratorPath=/absolute/path/to/artifacts/bin/AutoRest.CSharp/Debug/net6.0/AutoRest.CSharp.dll`
> 
> Inside the script we should wrap this using the existing methods to get the emitter name.


I update to remove emitter name in the parameter, just need to set key-value pair.